third_party/terraform/tests/resource_google_folder_iam_audit_config_test.go

```diff
@@ -0,0 +1,436 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func folderIamAuditConfigImportStep(t *testing.T, resourceName, fname, service string) resource.TestStep {
+	name, err := getFolderNameByParentAndDisplayName("organizations/"+getTestOrgFromEnv(t), fname, googleProviderConfig(t))
```

Comment:
It's complaining the *Config returned by googleProviderConfig(t) is {}, but I'm not sure why or how I can get the *Config and use it to retrieve the folder ID and use it in the Import check.

Reply:
Ok, so what is happening here is that there is a little confusion on the context that this is being evaluated in.

Generally `googleProviderConfig(t)` gives you the *Config that is being used for the test context that you are in. If you call it outside of a go test it will return {} because the *Config is only instantiated at the beginning of the test run, when `vcrTest(...)` is called. So you'll notice that all of the uses of `googleProviderConfig(t)` only exist in functions that are called from within the terraform process under test.

For example: https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_google_folder_iam_policy_test.go#L89

The call to `googleProviderConfig(t)` is within a `resource.TestCheckFunc`, so it gets evaluated within the `vcrTest` call. 

In your code, `folderIamAuditConfigImportStep` is evaluated *before* `vcrTest` is called, and passed as an argument to it. This causes an issue because:
The config doesn't exist
The folder doesn't exist, as no steps have run yet

I don't believe creating the import steps like this will work because of these issues.

Reply:
Do you have any recommendation around how to test this? I was trying to getting the folder ID from the display name and parent as for folder, the folder ID is needed in the terraform import. It seems it's not easy to get that piece of information.

Reply:
I removed the import state check for now.